cpp/daal/src/data_management/train_test_split.cpp

```diff
@@ -39,6 +40,59 @@ typedef daal::data_management::NumericTable::StorageLayout NTLayout;
 const size_t BLOCK_CONST      = 2048;
 const size_t THREADING_BORDER = 8388608;
 
+template <typename IdxType, daal::CpuType cpu>
+services::Status generateShuffledIndicesImpl(const NumericTablePtr & idxTable, const unsigned int seed)
```

Comment:
Please, provide profiling for following blocks of code to understand optimizations opportunities:
1) 46-56
2) 58 - 69
3) 71 - 74
4) full function